data-raw/northern-ireland-hsct-secondary-care.R

```diff
@@ -0,0 +1,147 @@
+# Northern Ireland - Secondary Care Indicators
+# Health and Social Care Trusts
+
+# ---- Load libs & helpers ----
+library(tidyverse)
+library(geographr)
+library(sf)
+library(healthyr)
+library(demographr)
+library(compositr)
+library(ggridges)
+library(devtools)
+library(readODS)
+
+hsct <- boundaries_trusts_ni18 |>
+  st_drop_geometry()
+
+lookup_trusts <- geographr::boundaries_trusts_ni18 |>
+  st_drop_geometry()
+
+# Population counts
+# Source: https://www.ninis2.nisra.gov.uk/public/AreaProfileReportViewer.aspx?FromAPAddressMulipleRecords=Northern@@Northern@20?
+
+url <- "https://www.ninis2.nisra.gov.uk/Download/Population/Population%20Estimates%20Broad%20Age%20Bands%20(administrative%20geographies).ods"
+
+pop_raw <- download_file(url, ".ods")
+
+pop_ni_hsct <- read_ods(
+  pop_raw,
+  sheet = "HSCT",
+  range = "A4:C9"
+) |>
+  select(
+    trust18_name = "HSCT",
+    trust18_code = "HSCT Code",
+    population = "Persons"
+  )
+
+# ---- RTT ----
+# Higher = worse performance
+rtt <- ni_rtt_hsct |>
+  select(
+    trust18_name = "hsct22_name",
+    date,
+    waits_over_18_weeks
+  ) |>
+  filter(date >= max(date) %m-% months(2)) |> # Last quarter
+  left_join(pop_ni_hsct) |>
+  mutate(percent = waits_over_18_weeks / population) |>
+  group_by(trust18_name) |>
+  summarise(
+    number = sum(waits_over_18_weeks),
+    percent = sum(percent, na.rm = FALSE)
+  ) |>
+  mutate(
+    variable = "Referral to treatment \nwaiting times (Mar 19 - Jun 22)",
+    .after = trust18_name
+  ) |>
+  filter(!trust18_name %in% c("DPC", "Day Case Procedure Centre"))
+
+# ---- Beds ----
+# Higher = better performance
+# without population for percentage
+available_beds <- ni_beds |>
+  left_join(lookup_trusts, by = "trust18_code") |>
+  mutate(percent_available = ((total_available_beds - total_occupied_beds) /
```

Comment:
Please can you check in the documentation that came with this data set (note, I don't mean the documentation you built for healthyr), whether "total available beds" means "the number of beds available regardless of how many are filled", or "the number of beds available after being filled at this given time point". If the former, your calculations seem correct.

Reply:
This is how average available beds was defined (there was no total available beds definition):
"Average Available/Occupied Beds: The average number of available and occupied beds during the year in wards that are open overnight, measured at midnight. Hospitals may also have a number of beds in wards that are only open during the day. Beds reserved for day care admission or regular day admission are not included."
I am not too sure but this does not seem to tell me much and does not clarify. If preferred, I can switch to the |average available/occupied beds" data  in the dataset?